src/SFA.DAS.ApplyService.Web/ViewModels/Roatp/TaskListViewModel.cs

```diff
@@ -157,5 +125,23 @@ public string DescribeOrganisationStartPageId
                 return describeOrganisationStartPageId;
             }
         }
+
+        public bool ApplicationSequencesCompleted()
+        {
+            var applicationSequences = ApplicationSequences.Where(seq => seq.SequenceId != RoatpWorkflowSequenceIds.Finish);
+            foreach (var sequence in applicationSequences)
+            {
+                foreach(var section in sequence.Sections)
+                {
+                    var sectionStatus = SectionStatus(sequence.SequenceId, section.SectionId).ToLower();
```

Comment:
are you able to use section.SequenceId and flatten this foreach loop?

Reply:
No, the hierarchy is Sequences -> Sections and it needs to iterate through all to ensure that everything mandatory has been answered

Reply:
I was thinking more on the lines of: applicationSequences.SelectMany(as => as.Sections)